fedimint-bitcoind/src/bitcoincore_rpc.rs

```diff
@@ -162,3 +180,76 @@ where
         })
     }
 }
+
+pub struct ElectrumClient(electrum_client::Client);
+
+impl ElectrumClient {
+    fn new(url: &Url) -> anyhow::Result<Self> {
+        Ok(Self(electrum_client::Client::new(url.as_str())?))
+    }
+}
+
+impl fmt::Debug for ElectrumClient {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str("ElectrumClient")
+    }
+}
+use bitcoin_hashes::hex::ToHex;
+use electrum_client::ElectrumApi;
+
+#[async_trait]
+impl IBitcoindRpc for ElectrumClient {
+    async fn get_network(&self) -> Result<Network> {
+        let resp = fedimint_api::task::block_in_place(|| {
+            self.0.server_features().map_err(anyhow::Error::from)
+        })?;
+        Ok(match resp.genesis_hash.to_hex().as_str() {
+            "000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f" => Network::Bitcoin,
+            n => panic!("Unknown Genesis Hash \"{}\"", n),
+        })
```

Comment:
I wonder if we can do better with the tools rust-bitcoin gives us? At least we could do a linear search over networks and their chain hashes iirc?